Mf2/Parser.php

```diff
@@ -663,73 +682,96 @@ public function parseDT(\DOMElement $dt, &$dates = array()) {
 			foreach ($valueClassChildren as $e) {
 				if (strstr(' ' . $e->getAttribute('class') . ' ', ' value-title ')) {
 					$title = $e->getAttribute('title');
-					if (!empty($title))
+					if (!empty($title)) {
 						$dateParts[] = $title;
+					}
 				}
 				elseif ($e->tagName == 'img' or $e->tagName == 'area') {
 					// Use @alt
 					$alt = $e->getAttribute('alt');
-					if (!empty($alt))
+					if (!empty($alt)) {
 						$dateParts[] = $alt;
+					}
 				}
 				elseif ($e->tagName == 'data') {
 					// Use @value, otherwise innertext
 					$value = $e->hasAttribute('value') ? $e->getAttribute('value') : unicodeTrim($e->nodeValue);
-					if (!empty($value))
+					if (!empty($value)) {
 						$dateParts[] = $value;
+					}
 				}
 				elseif ($e->tagName == 'abbr') {
 					// Use @title, otherwise innertext
 					$title = $e->hasAttribute('title') ? $e->getAttribute('title') : unicodeTrim($e->nodeValue);
-					if (!empty($title))
+					if (!empty($title)) {
 						$dateParts[] = $title;
+					}
 				}
 				elseif ($e->tagName == 'del' or $e->tagName == 'ins' or $e->tagName == 'time') {
 					// Use @datetime if available, otherwise innertext
 					$dtAttr = ($e->hasAttribute('datetime')) ? $e->getAttribute('datetime') : unicodeTrim($e->nodeValue);
-					if (!empty($dtAttr))
+					if (!empty($dtAttr)) {
 						$dateParts[] = $dtAttr;
+					}
 				}
 				else {
-					if (!empty($e->nodeValue))
+					if (!empty($e->nodeValue)) {
 						$dateParts[] = unicodeTrim($e->nodeValue);
+					}
 				}
 			}
 
 			// Look through dateParts
 			$datePart = '';
 			$timePart = '';
+			$timezonePart = '';
 			foreach ($dateParts as $part) {
 				// Is this part a full ISO8601 datetime?
-				if (preg_match('/^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}(?::\d{2})?(?:Z?[+|-]\d{2}:?\d{2})?$/', $part)) {
+				if (preg_match('/^\d{4}-\d{2}-\d{2}T?\d{2}:\d{2}(?::\d{2})?(?:Z?[+|-]\d{2}:?\d{2})?$/', $part)) {
```

Comment:
Why the added `?` here? That makes the `T` optional. Should this instead be checking for a ` ` as well as `T`?